src/Form/Form.js

```diff
@@ -18,6 +18,18 @@ function findFieldOption(options, field) {
   });
 }
 
+function mergeNewModel(stateModel, currentModelDefinition, nextModelDefinition) {
+  let newModel = {};
+
+  Object.keys(nextModelDefinition).forEach(function (key) {
+    if (currentModelDefinition[key] !== nextModelDefinition[key]) {
```

Comment:
This means that it will override if the new property is `undefined` or `null` too. As long as that is what we want :)